.changeset/modern-melons-clean.md

```diff
@@ -0,0 +1,32 @@
+---
+"@graphql-tools/merge": patch
+"@graphql-tools/stitch": minor
+"@graphql-tools/website": patch
+---
+
+Introduces a suite of stitched schema validations that enforce the integrity of merged schemas. This includes validations for:
+
+- Strict and safe null consistency (the later of which allows safe transitions in nullability).
+- Named type consistency with the option to whitelist proxiable scalar mappings.
+- Argument and input field name consistency.
+- Enum value consistency when used as an input value.
+
+Validations may be adjusted by setting `validationLevel` to `off|warn|error` globally or for specific field scopes. In this initial v7 release, all validations are introduced at the `warn` threshold for backwards compatibility. Most of these validations will become automatic errors in v8. To enable validation errors now, set `validationLevel: 'error'`. Full configuration options look like this:
+
+```js
+const gatewaySchema = stitchSchemas({
+  subschemas: [...],
+  typeMergingOptions: {
+    validationSettings: {
+      validationLevel: 'error',
+      strictNullComparison: false, // << gateway "String" may proxy subschema "String!"
+      proxiableScalars: {
+        ID: ['String'], // << gateway "ID" may proxy subschema "String"
+      }
+    },
+    validationScopes: {
+      'User.id': { strictNullComparison: true },
```

Comment:
Just a small thought to include here that you can change the validationLevel on a per-scope basis, right? that is really great!

Reply:
Sound okay? 44e41ba